Sources/GRPC/AsyncAwaitSupport/GRPCAsyncRequestStreamWriter.swift

```diff
@@ -72,6 +66,22 @@ public struct GRPCAsyncRequestStreamWriter<Request: Sendable>: Sendable {
     try await self.asyncWriter.yield((request, compression))
   }
 
+  /// Send a sequence of requests.
+  ///
+  /// It is safe to send multiple requests concurrently by sharing the ``GRPCAsyncRequestStreamWriter`` across tasks.
+  ///
+  /// Callers must call ``finish()`` when they have no more requests left to send.
+  ///
+  /// - Parameters:
+  ///   - contentsOf: The requests and their compression to send.
+  /// - Throws: If the request stream has already been finished.
+  @inlinable
+  public func send<S: Sequence>(
+    contentsOf requests: S
+  ) async throws where S.Element == (Request, Compression) {
+    try await self.asyncWriter.yield(contentsOf: requests)
+  }
```

Comment:
@Lukasa @fabianfett What do you think about using the tuple here? Do you prefer creating a new type like  `MessageWithCompression`?

Reply:
Fine either way. But generally structs are the better choice for evolving types and APIs. Name should probably be `MessageAndCompression`? With sounds like the message was already compressed.

Reply:
Actually I changed the API a bit for now where you can write a sequence of requests/responses and set the compression for all of them. If users ever want to write a sequence with different compression per request/response we can still add that API. WDYT?